tests/golden_tests.rs

```diff
@@ -169,3 +170,23 @@ fn encode_pattern_match() {
     Ok(book.to_string())
   })
 }
+
+#[test]
+#[ignore = "To not delay golden tests execution"]
+fn hangs() {
+  let expected_normalization_time = 1;
+
+  run_golden_test_dir(function_name!(), &|code| {
+    let book = do_parse_book(code)?;
+
+    let lck = Arc::new(RwLock::new(false));
+    let got = lck.clone();
+    std::thread::spawn(move || {
+      let _ = run_book(book, 1 << 20, true, false, false, true, Opts::heavy());
+      *got.write().unwrap() = true;
+    });
+    std::thread::sleep(std::time::Duration::from_secs(expected_normalization_time));
+
+    if !*lck.read().unwrap() { Err("Hangs".into()) } else { Ok("Not hangs".into()) }
```

Comment:
Isn't this for files we expect to hang? So Ok and Err should be swapped.
Also, "Doesn't hang".